src/include/sof/trace/trace.h

```diff
@@ -170,11 +170,22 @@ void trace_log(bool send_atomic, const void *log_entry,
 unsupported_amount_of_params_in_trace_event\
 _thrown_from_macro_BASE_LOG_in_trace_h
 
+
+#define CT_ASSERT(COND, MESSAGE) \
+	((void)sizeof(char[1 - 2*!(COND)]))
+
+#define trace_check_size_uint32(a) \
+	CT_ASSERT(sizeof(a) <= sizeof(uint32_t), "error: trace argument is bigger than a uint32_t");
```

Comment:
wouldn't `STATIC_ASSERT()` work here?

Reply:
`STATIC_ASSERT()` implementation doesn't allow multiple assertion with the same message, what is needed here.